Implement argument passing for lat/lon starting position

oli-obk opened the pull request:
cc https://github.com/DerKarlos/OSMeta/issues/29

DerKarlos reviewed (changes requested):
I am surprised, how much of my code you used already, even the module cam_map!.
My osm_scene.rs is about your map.rs, but should not be used; may be some parts of it.
What about geo_view? I used it at boot time and to store places (default and dynamical, user defined)

util.rs is ok. Someone at Youtube told me: don’t use this helper sources, put the values, where they rise. I intended it. That may be not that easy.


The term "TileName” raised historical at OSM. It was discussed by me and Tobias. I would be happy to replace it by TileCoorinates (or TileCoors?).
Just using Vec2 instead would the code make less clear. Examples: 
pub fn to_tile_coordinates(&self, zoom: u8) -> Vec2 {
Vec2 { x, y } = pos.to_tile_coordinates(15);

The coordinates are u32 but f32 is ok also and will the code make more readable.

TileMap<8145>. The unit dm decimetre is a bit odd? I learned to always use IOS units; m in this case by f32.

oli-obk commented:
> TileMap<8145>. The unit dm decimetre is a bit odd? I learned to always use IOS units; m in this case by f32.

We can't use f32 in constants yet, but I also have a local branch that should allow us to remove the constant entirely and compute the tile height/width from lat/lon .



> The term "TileName” raised historical at OSM. It was discussed by me and Tobias. I would be happy to replace it by TileCoorinates (or TileCoors?).
> Just using Vec2 instead would the code make less clear. Examples:
> pub fn to_tile_coordinates(&self, zoom: u8) -> Vec2 {
> Vec2 { x, y } = pos.to_tile_coordinates(15);

Yea, i'll add a wrapper type that makes it clear



> util.rs is ok. Someone at Youtube told me: don’t use this helper sources, put the values, where they rise. I intended it. That may be not that easy.

I removed it since it was not needed yet.



> My osm_scene.rs is about your map.rs, but should not be used; may be some parts of it.
> What about geo_view? I used it at boot time and to store places (default and dynamical, user defined)

I removed both for now, but we'll def want to use parts of osmscene in the tile_map. The view being remembered across application restarts is a good idea. Though I think there are some bevy "savegame" plugins we can try out. I'll open an issue



> I am surprised, how much of my code you used already, even the module cam_map!.

Most of it got removed again in later commits.

> The coordinates are u32 but f32 is ok also and will the code make more readable.

I did this on purpose. We use u32 for tile coordinates, but f32 for the transformation. This way we can go from GPS coordinates to positions *within* a tile and make the camera look at individual buildings instead of a tile corner.

DerKarlos commented:
> Yea, i'll add a wrapper type that makes it clear
💚
